src/raytracing.jl

```diff
@@ -70,18 +71,74 @@ normal_to_barrel(c::Cylinder, p:: Vector{Float64}) = [p[1], p[2], 0.0] ./c.r
 
 equation of cylynder: F(x,y,z) = x^2 + y^2 - r^2 = 0
 """
-cylinder_equation(c::Cylinder, p:: Vector{Float64}) = p[1]^2 + p[2]^2 - c.r^2
+cylinder_equation(c::Cylinder, p::Vector{Float64}) = p[1]^2 + p[2]^2 - c.r^2
 
 
-length(c::Cylinder) = abs.(c.zmax - c.zmin)
+cylinder_length(c::Cylinder) = abs.(c.zmax - c.zmin)
     
 perimeter(c::Cylinder) = 2 * π * c.r
 
-area_barrel(c::Cylinder) = 2 * π * c.r * length(c)
+area_barrel(c::Cylinder) = 2 * π * c.r * cylinder_length(c)
     
 area_endcap(c::Cylinder) = π * c.r^2
     
 area(c::Cylinder) = area_barrel(c) + 2 * area_endcap(c)
 
-volume(c::Cylinder)= area_endcap(c) * length(c)
+volume(c::Cylinder)= area_endcap(c) * cylinder_length(c)
+
+
+"""Returns True if point in end-caps of cyinder"""
+in_endcaps(c:: Cylinder, p::Vector{Float64}) = isapprox(p[3], c.zmin) || isapprox(p[3], c.zmax)
```

Comment:
This alone doesn't test if the point is in an endcap. Only that the one-dimensional position is level with the endcaps

Reply:
That is what was intended, name probably not very well chosen